mobx_codegen/test/util_test.dart

```diff
@@ -12,16 +12,16 @@ class MockMethod extends Mock implements MethodElement {}
 
 class MockType extends Mock implements DartType {}
 
+class DynamicMockType extends Mock implements DynamicType {}
+
 MockMethod mockFutureMethod({
   bool returnsDynamic = false,
   bool returnsFuture = false,
   bool returnsFutureOr = false,
   bool isAsync = false,
   bool isGenerator = false,
 }) {
-  final returnType = MockType();
-  // ignore: deprecated_member_use
-  when(() => returnType.isDynamic).thenReturn(returnsDynamic);
+  final returnType = returnsDynamic ? DynamicMockType() : MockType();
```

Comment:
Change to: (as mentioned by @davidmartos96 in https://github.com/mobxjs/mobx.dart/pull/1038/files#r1981540688)

```suggestion
  final DartType returnType;
  if (returnsDynamic) {
    returnType = DynamicMockType();
  } else {
    returnType = MockType();
  }
```